src/ui/public/agg_types/metrics/cumulative_sum.js

```diff
@@ -0,0 +1,17 @@
+import AggTypesMetricsMetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import ParentPipelineAggHelperProvider from './lib/parent_pipeline_agg_helper';
+import { makeNestedLabel } from './lib/make_nested_label';
+
+export default function AggTypeMetricComulativeSumProvider(Private) {
+  const MetricAggType = Private(AggTypesMetricsMetricAggTypeProvider);
+  const parentPipelineAggHelper = Private(ParentPipelineAggHelperProvider);
+
+  return new MetricAggType({
+    name: 'cumulative_sum',
+    title: 'Cumulative Sum',
+    makeLabel: agg => makeNestedLabel(agg, 'cumulative sum'),
+    params: [
+      ...parentPipelineAggHelper.params()
```

Comment:
so yeah, its pretty much the same, but different ES query will get executed (the name defines the name used in ES query .... so this one will run cumulative_sum instead of derivative ... but apart from that they are the same.